pkg/cableengine/syncgatewaystatus.go

```diff
@@ -0,0 +1,161 @@
+package cableengine
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"strconv"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+
+	v1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/log"
+)
+
+const GatewayUpdateIntervalSeconds = 5
+const GatewayStaleTimeoutSeconds = GatewayUpdateIntervalSeconds * 3
+const updateTimestampAnnotation = "update-timestamp"
+
+func (i *engine) syncGatewayStatus() {
+
+	i.Lock()
+	defer i.Unlock()
+
+	klog.V(log.TRACE).Info("Running syncGatewayStatus()")
+
+	gatewayObj, err := i.generateGatewayObject()
+	if err != nil {
+		klog.Errorf("generating gateway object from driver connections: %s", err)
+		return
+	}
+
+	existingGw, err := i.getLastSyncedGateway(gatewayObj.Name)
+
+	// log and stop for any error different to a not-found error
+	if err != nil && !errors.IsNotFound(err) {
+		klog.Errorf("trying to read existing gateway from k8s api: %s", err)
+		return
+	}
+
+	gwClient := i.clientset.SubmarinerV1().Gateways(i.namespace)
+
+	if err != nil && errors.IsNotFound(err) {
+		klog.V(log.TRACE).Infof("Gateway object needs creation: %+v", gatewayObj)
+		_, err = gwClient.Create(gatewayObj)
+		if err != nil {
+			klog.Errorf("creating Gateway object: %s", err)
```

Comment:
Can you please take a look at the Errorf log messages in the patch and include "error " as part of the message.
i.e., "error creating Gateway object: ...."
I know we are printing the err as part of it. While debugging some problem and looking at a long log message, it would be easy if the error messages starts with "error".

Reply:
I thought that's an unwanted practice because the log will be already indicative of error, may be I got it wrong?

Reply:
E0417 13:29:03.745481       1 syncgatewaystatus.go:49] creating Gateway object: the server could not find the requested resource (post gateways.submariner.io)


Oh you're right, it's prefixed by "E" , but that's not very helpful for human readers